app/services/finance/stripe_charge_service.rb

```diff
@@ -75,4 +80,12 @@ def extract_payment_intent_data_from(response)
     payment_intent_data = (response.success? ? response_params : response_params.dig('error', Stripe::PaymentIntent::OBJECT_NAME)) || {}
     payment_intent_data if payment_intent_data['object'] == Stripe::PaymentIntent::OBJECT_NAME
   end
+
+  def charge_description
+    if @invoice
+      "#{invoice.try(:from)&.name} #{PAYMENT_DESCRIPTION} #{invoice.try(:friendly_id)}"
+    else
+      PAYMENT_DESCRIPTION
+    end
+  end
```

Comment:
A small suggestion to change this method. Main motivation was to remove the "try(:something)&.something_else" so we use the same syntax in both tries, as I think we only want to safely navigate through nils. 

```suggestion
  def charge_description
    return PAYMENT_DESCRIPTION unless @invoice.present? 
    
    "#{invoice&.from&.name} #{PAYMENT_DESCRIPTION} #{invoice&.friendly_id}".strip
  end
```